webpack/components/WithOrganization/__snapshots__/withOrganization.test.js.snap

```diff
@@ -1,41 +1,175 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`subscriptions page should render select org page 1`] = `
-<CheckOrg>
-  <Header
-    title="Select Organization"
+<Connect(CheckOrg)
```

Comment:
Those snapshots became too big to maintain and understand.

I would like to suggest a solution to it:

1. Separate the `CheckOrg` class from the `withOrganization.js` to a new file (`CheckOrg.js`) so it will be a standalone component.
The `CheckOrg` component will recive a children instead using the `WrappedComponent`.
The `withOrganization` hoc will render `CheckOrg` with the `WrappedComponent` as it's children and will connect it to `redux`.
2. Do the current snapshoot  testings only for the `CheckOrg` component so we will have lean snapshoots.
3. Need to have lean snapshoot testing for actions/reducers/selectors/helpers.
4. Need to have a lean integration-testing for the HOC that run all the react-redux lifecycle using the [IntegrationTestHelper](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/common/IntegrationTestHelper.js).
Notice the `IntegrationTestHelper` is in a process to be a standalone npm package, see https://github.com/theforeman/foreman/pull/6432 and https://github.com/Katello/katello/pull/7940
You can use those docs: https://sharvit.github.io/react-redux-test-utils/manual/integration-testing.html
And some examples:
https://github.com/theforeman/foreman/blob/edc0a1c73eb59b28b266e63a1a028c788837a942/webpack/assets/javascripts/react_app/components/Layout/__tests__/integration.test.js
https://github.com/theforeman/foreman/blob/c18d67464d81943b77feeb0e24ec649574860ab7/webpack/assets/javascripts/react_app/components/BreadcrumbBar/__tests__/integration.test.js
https://github.com/theforeman/foreman/blob/26345fcd991a3d3ca63f917ae1eed161167dd5aa/webpack/assets/javascripts/react_app/components/SearchBar/__tests__/integration.test.js
https://github.com/theforeman/foreman/blob/c18d67464d81943b77feeb0e24ec649574860ab7/webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/integration.test.js

What do you think?

Reply:
I get what you are thinking, but I'm having a really hard time putting this into practice.

Here is where I am at https://github.com/johnpmitsch/katello/commit/5b746864d65f764e1b8b25c7dcf9f68817b5ae10 I've had to work past some issues including things like [HOCs not playing nice with react-router](https://github.com/ReactTraining/react-router/issues/6056)

The current error that i'm blocked by is:
```
react-dom.development.js:506 Warning: Functions are not valid as a React child. This may happen if you return a Component instead of <Component /> from render. Or maybe you meant to call this function rather than return it.
    in Unknown (created by Route)
    in Route
    in div
    in Unknown (created by Application)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by Application)
    in Application (created by Connect(Application))
    in Connect(Application) (created by I18nProviderWrapper(Connect(Application)))
    in IntlProvider (created by I18nProviderWrapper(Connect(Application)))
    in I18nProviderWrapper(Connect(Application)) (created by StoreProvider(I18nProviderWrapper(Connect(Application))))
    in Provider (created by StoreProvider(I18nProviderWrapper(Connect(Application))))
    in StoreProvider(I18nProviderWrapper(Connect(Application))) (created by DataProvider(StoreProvider(I18nProviderWrapper(Connect(Application)))))
    in DataProvider(StoreProvider(I18nProviderWrapper(Connect(Application))))```

Let me know if you can see where things are going wrong, I'll keep looking, but i'm pretty stumped to be honest :)

Reply:
Too bad, code-wise it makes so much sense :disappointed: